content/en/docs/guides/security/credential_reloading_file_watcher.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Credential Reloading on File Changes
+description: >-
+  An overview of how to enable an experimental feature, credential reloading on 
+  file changes, in gRPC C++, Java and Go.
+---
+
+### Overview
+
+`AdvancedTLS` is a set of security features we wanted to introduce to our 
+open-source users who want more advanced controls over their TLS connections. 
+Credential reloading is one important feature in AdvancedTLS.
+
+When we refer to the credential, we usually mean the following three pieces:
+1. private key, which should be kept in a safe place,
+2. identity certificate chain, which is sent to the peers to prove endpoint's 
+identity, and
+3. trust certificate bundle, which basically tells which CA to trust
+
+In normal mutual TLS connections, once the credential set is configured, there 
+is no easy way to change it without shutting down the server, or re-creating the 
+client. This will create some problems if we want a long-lived server, or we 
+want to start a long-lived connection from the client, because the certificates 
+can expire and certificate rotations are required.
+
+To solve this problem, we introduced a new set of API to reload credentials from 
+disk. To use it, users just need to specify the absolute file paths for the 
+credentials when creating their client/server. gRPC will monitor the changes of 
+those files, and update them when necessary. When there is a need to use new 
+credentials, simply just change the file contents on disk, while the gRPC 
+server/client is running. gRPC will make sure the operation is thread-safe.
```

Comment:
Make sure which operation is thread-safe?

I think that's probably an unnecessary detail though. It seems better to say something like "When gRPC notices the files are changed, it will reload the credentials and use them on future connections."

Reply:
Cool, that's better than my original wording. Changed accordingly.